File: boxsys/FullViewController.m
//
//  FullViewController.m
//  boxsys
//
//  Created by chunyi.zhoucy on 14-3-6.
//  Copyright (c) 2014年 chunyi.zhoucy. All rights reserved.
//

#import "FullViewController.h"
#import "AppDelegate.h"

@interface FullViewController ()

@end

@implementation FullViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    NSDictionary *dData = [AppDelegate getDictionaryFromFile:@"DataFullViewController"];
    if (!dData) return;
    [JuBoxSys2 loadBox:@"Today"
               withKey:@"ViewController"
              withData:dData
                 width:320 height:0 withDelegate:self];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (UIView *) getContainer
{
    return self.view;
}

- (void) onBoxSysEvent:(id) sender withArgs:(NSDictionary *) argsMap
{
    
}

@end

File: boxsys/FullViewController.swift
//
//  FullViewController.swift
//  boxsys
//
//  Migrated to Swift from Objective-C
//

import UIKit

class FullViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, JuBoxSysDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    // MARK: - UITableViewDataSource

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 0
    }

    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        return UITableViewCell()
    }

    // MARK: - JuBoxSysDelegate

    func getContainer() -> UIView {
        return self.view
    }

    func onBoxSysEvent(sender: Any, args: [String: Any]?) {
        // Handle box system events
    }
}
